### PR TITLE
Allow to load file directly into the VFS

### PR DESCRIPTION
**Pull request by matklad**
This is the scaled down version of https://github.com/nrc/rls-vfs/pull/7. 

Does this look reasonable? I'll update RSL as well if this is OK :) 

PS: am I correct that we don't use `rustfmt` for this repository yet?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -30,9 +30,26 @@ pub struct Vfs<U = ()>(VfsInternal<RealFileLoader, U>);
...
-    pub span: Span,
-    pub text: String,
+pub enum Change {
+    /// Create an in-memory image of the file.
+    AddFile {
+        file: PathBuf,
+        text: String,
+    },
+    /// Changes in-memory contents of the previously added file.
+    Replace {
```
Comment (nrc) [inline src/lib.rs]: Maybe ReplaceText for symmetry with AddFile

File: src/lib.rs [hunk 2]
```diff
@@ -224,6 +241,12 @@ impl<T: FileLoader, U> VfsInternal<T, U> {
                 }
             }
 
+            // XXX: if the first change is `Add`, we should avoid loading
```
Comment (nrc) [inline src/lib.rs]: Could you file an issue for this and s/XXX/FIXME(#...)

**Review by nrc — CHANGES_REQUESTED** [review]
LGTM, thanks! Just a couple of nits to address

**Comment by matklad** [comment]
Nits fixed, here's the RLS PR: https://github.com/jonathandturner/rls/pull/148

**Comment by nrc** [comment]
Great, thank you!

**Pull request merged**